src/helpers/redisHelper.ts

```diff
@@ -0,0 +1,89 @@
+import { StoredUrl, UrlWithInformation } from '../services/storage/types/url.js'
+import { RedisStorage } from '../services/storage/drivers/redis'
+import { Tedis } from 'tedis'
+import { NotFoundError } from '../errors/notFound.js'
+import { InvalidPayloadError } from '../errors/invalidUrl.js'
+
+export class RedisHelper {
+	private redis: Tedis
+	constructor(storage: RedisStorage) {
+		this.redis = storage.getRedisClient()
+	}
+	url: StoredUrl = {
+		id: '',
+		url: '',
+		createdAt: '',
+		updatedAt: '',
+	}
+	urlWithInfo: UrlWithInformation = {
+		id: '',
+		urlId: '',
+		url: '',
+		createdAt: '',
+		updatedAt: '',
+		ip: '',
+		urlVisitCount: 0,
+		infoVisitCount: 0,
+		lastUsed: '',
+	}
+
+	async fetchUrlInfoFromDB(id: string, options: { withInfo: boolean }): Promise<StoredUrl | UrlWithInformation> {
+		const rawUrlData = (await this.redis.get(id)) as string
+		if (!rawUrlData)
+			throw new NotFoundError()
+		this.processUrlFromStorage(rawUrlData)
+		this.urlWithInfo.urlId = id
+		if (!options.withInfo) {
+			return this.url
+	}
+		else {
+			return this.urlWithInfo
+		}
+	}
+
+	generateUrl(key: string, value: string): void{
```

Comment:
can you abstract this method? - remove the switch case and use more abstract method